src/AspNet.Security.OAuth.SuperOffice/SuperOfficeAuthenticationHandler.cs

```diff
@@ -125,27 +112,29 @@ protected new SuperOfficeAuthenticationEvents Events
             return new AuthenticationTicket(context.Principal, context.Properties, Scheme.Name);
         }
 
-        /// <summary>
-        /// Extracts the identity claims from the token received from the token endpoint.
-        /// </summary>
-        /// <param name="token">The token to extract the claims from.</param>
-        /// <returns>
-        /// An <see cref="IEnumerable{Claim}"/> containing the claims extracted from the token.
-        /// </returns>
-        protected virtual IEnumerable<Claim> ExtractClaimsFromToken([NotNull] string token)
+        private async Task<ClaimsPrincipal> ValidateAsync(string idToken)
         {
-            try
+            if (!_tokenHandler.CanValidateToken)
             {
-                var securityToken = _tokenHandler.ReadJwtToken(token);
+                throw new NotSupportedException($"The configured {nameof(JwtSecurityTokenHandler)} cannot validate tokens.");
+            }
 
-                return new List<Claim>(securityToken.Claims)
-                {
-                    new Claim(ClaimTypes.NameIdentifier, securityToken.Subject, ClaimValueTypes.String, ClaimsIssuer),
-                };
+            if (Options.ConfigurationManager == null)
+            {
+                throw new InvalidOperationException($"An OpenID Connect configuration manager has not been set on the {nameof(SuperOfficeAuthenticationOptions)} instance.");
+            }
+
+            var openIdConnectConfiguration = await Options.ConfigurationManager.GetConfigurationAsync(Context.RequestAborted);
+            Options.TokenValidationParameters.IssuerSigningKeys = openIdConnectConfiguration.JsonWebKeySet.Keys;
+
+            try
+            {
+                return _tokenHandler.ValidateToken(idToken, Options.TokenValidationParameters, out var _);
             }
             catch (Exception ex)
             {
-                throw new InvalidOperationException("Failed to parse JWT for claims from SuperOffice ID token.", ex);
+                throw new SecurityTokenValidationException(
+                    "SuperOffice ID token validation failed for issuer {TokenIssuer} and audience {TokenAudience}.", ex);
```

Comment:
Missing string interpolation:

```suggestion
                    $"SuperOffice ID token validation failed for issuer {TokenIssuer} and audience {TokenAudience}.", ex);
```